diff --git a/configure b/configure
index d88c47c..8ff85f6 100755
--- a/configure
+++ b/configure
@@ -4578,6 +4578,10 @@ aix[4-9]*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -20658,7 +20662,7 @@ else
 esac
 
 
-$as_echo "#define malloc rpl_malloc" >>confdefs.h
+#$as_echo "#define malloc rpl_malloc" >>confdefs.h
 
 fi
 
@@ -20729,7 +20733,7 @@ else
 esac
 
 
-$as_echo "#define realloc rpl_realloc" >>confdefs.h
+#$as_echo "#define realloc rpl_realloc" >>confdefs.h
 
 fi
 
