diff --git a/src/filter.c b/src/filter.c
index 71f3635..7a9761e 100644
--- a/src/filter.c
+++ b/src/filter.c
@@ -163,7 +163,7 @@ bool filter_apply_chain (struct filter * chain)
          */
 		close (pipes[1]);
 clearerr(stdin);
-		if (dup2 (pipes[0], fileno (stdin)) == -1)
+		if (dup2 (pipes[0], STDIN_FILENO) == -1)
 			flexfatal (_("dup2(pipes[0],0)"));
 		close (pipes[0]);
         fseek (stdin, 0, SEEK_CUR);
@@ -190,7 +190,7 @@ clearerr(stdin);
 
 	/* Parent */
 	close (pipes[0]);
-	if (dup2 (pipes[1], fileno (stdout)) == -1)
+	if (dup2 (pipes[1], STDOUT_FILENO) == -1)
 		flexfatal (_("dup2(pipes[1],1)"));
 	close (pipes[1]);
     fseek (stdout, 0, SEEK_CUR);
diff --git a/src/misc.c b/src/misc.c
index ef27833..85d7035 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -300,6 +300,7 @@ void flexerror (const char *msg)
 
 void flexfatal (const char *msg)
 {
+  perror("failed");
 	fprintf (stderr, _("%s: fatal internal error, %s\n"),
 		 program_name, msg);
 	FLEX_EXIT (1);
